gobblin-yarn/src/main/java/gobblin/yarn/GobblinYarnAppLauncher.java

```diff
@@ -0,0 +1,598 @@
+      Path destFilePath = new Path(destDir, libJarFile.getPath().getName());
+      this.fs.copyFromLocalFile(libJarFile.getPath(), destFilePath);
+      if (resourceMap.isPresent()) {
+        addFileAsLocalResource(destFilePath, LocalResourceType.FILE, resourceMap.get());
+      }
+    }
+  }
+
+  private void addAppJars(String jarFilePathList, Optional<Map<String, LocalResource>> resourceMap,
+      Path destDir) throws IOException {
+    for (String jarFilePath : SPLITTER.split(jarFilePathList)) {
+      Path srcFilePath = new Path(jarFilePath);
+      Path destFilePath = new Path(destDir, srcFilePath.getName());
+      this.fs.copyFromLocalFile(srcFilePath, destFilePath);
+      if (resourceMap.isPresent()) {
+        addFileAsLocalResource(destFilePath, LocalResourceType.FILE, resourceMap.get());
+      }
+    }
+  }
+
+  private void addAppLocalFiles(String localFilePathList, Optional<Map<String, LocalResource>> resourceMap,
+      Path destDir) throws IOException {
+    for (String localFilePath : SPLITTER.split(localFilePathList)) {
+      Path srcFilePath = new Path(localFilePath);
+      Path destFilePath = new Path(destDir, srcFilePath.getName());
+      this.fs.copyFromLocalFile(srcFilePath, destFilePath);
+      if (resourceMap.isPresent()) {
+        addFileAsLocalResource(destFilePath, LocalResourceType.FILE, resourceMap.get());
+      }
+    }
+  }
+
+  private void addAppRemoteFiles(String hdfsFileList, Map<String, LocalResource> resourceMap)
+      throws IOException {
+    for (String hdfsFilePath : SPLITTER.split(hdfsFileList)) {
+      addFileAsLocalResource(new Path(hdfsFilePath), LocalResourceType.FILE, resourceMap);
+    }
+  }
+
+  private void addJobConfPackage(String jobConfPackagePath, Path destDir, Map<String, LocalResource> resourceMap)
+      throws IOException {
+    Path srcFilePath = new Path(jobConfPackagePath);
+    Path destFilePath = new Path(destDir, srcFilePath.getName());
+    this.fs.copyFromLocalFile(srcFilePath, destFilePath);
+    addFileAsLocalResource(destFilePath, LocalResourceType.ARCHIVE, resourceMap);
+  }
+
+  private void addFileAsLocalResource(Path destFilePath, LocalResourceType resourceType,
+      Map<String, LocalResource> resourceMap) throws IOException {
+    LocalResource fileResource = Records.newRecord(LocalResource.class);
+    FileStatus fileStatus = this.fs.getFileStatus(destFilePath);
+    fileResource.setResource(ConverterUtils.getYarnUrlFromPath(destFilePath));
+    fileResource.setSize(fileStatus.getLen());
+    fileResource.setTimestamp(fileStatus.getModificationTime());
+    fileResource.setType(resourceType);
+    fileResource.setVisibility(LocalResourceVisibility.APPLICATION);
+    LOGGER.debug(String.format("Created a LocalResource for file %s of type %s",
+        fileResource.getResource(), resourceType));
+    resourceMap.put(destFilePath.getName(), fileResource);
+  }
+
+  private Map<String, String> getEnvironmentVariables() {
```

Comment:
Again, this seems similar to `YarnClient. getEnvironmentVariables`. I would suggest moving all these methods out into a separate class.